### PR TITLE
Ci warnint v3

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
None

Describe changes:
- Makes CI error on integer warnings !
- Fixes one newly introduced integer warning (by eb155639c6293e5fd412a8268a2c1a6ca03ebe45)

libhtp-pr: 369

Modifies https://github.com/OISF/suricata/pull/8144 by adding debug validations in defrag code...

### CODE REVIEW

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on report_failure.

ERROR: QA failed on ASAN_TLPR1_suri.
  

Pipeline 10570

File: src/defrag.c [hunk 1]
```diff
@@ -336,7 +336,11 @@ Defrag4Reassemble(ThreadVars *tv, DefragTracker *tracker, Packet *p)
                     frag->data_len - frag->ltrim) == -1) {
                 goto error_remove_tracker;
             }
-            DEBUG_VALIDATE_BUG_ON(frag->offset <= UINT16_MAX - frag->data_len);
+            if (frag->offset <= UINT16_MAX - frag->data_len) {
+                SCLogWarning(SC_ERR_REASSEMBLY, "Failed re-assemble "
```
Comment (victorjulien) [inline src/defrag.c:340]: no output in the packet path

Comment (catenacyber) [reply src/defrag.c:340]: copied from above...

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPR1_stats_chk | | |
| .decoder.event.icmpv4.unknown_type | 920 | 27 | 2.93% |
| .decoder.event.icmpv4.unknown_code | 381 | 182 | 47.77% |
| .defrag.ipv4.reassembled | 57804 | 0 | - |
| .app_layer.flow.quic | 41209 | 31950 | 77.53% |
| .app_layer.tx.quic | 319598 | 287432 | 89.94% | 

Pipeline 10572

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW1_stats_chk | | |
| .tcp.rst | 128237 | 103975 | 81.08% |
| .defrag.ipv4.reassembled | 2 | 0 | - |
| | SURI_TLPR1_stats_chk | | |
| .decoder.event.icmpv4.unknown_type | 920 | 27 | 2.93% |
| .decoder.event.icmpv4.unknown_code | 381 | 182 | 47.77% |
| .defrag.ipv4.reassembled | 57804 | 0 | - |
| .app_layer.flow.quic | 41209 | 31950 | 77.53% |
| .app_layer.tx.quic | 319598 | 287432 | 89.94% | 

Pipeline 10581

**Comment by victorjulien** [comment]
Hmm the qa output suggests some issue with defrag in this PR. Wouldn't expect these fixes to change the behavior.

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/8168

**Pull request closed without merging**